modules/sdk-coin-atom/src/lib/utils.ts

```diff
@@ -0,0 +1,228 @@
+    return re.test(address);
+  }
+
+  /**
+   * Returns whether or not the string is a valid amount
+   *
+   * @param {number[]} amounts - the amounts to validate
+   * @returns {boolean} - the validation result
+   */
+  isValidAmounts(amounts: number[]): boolean {
+    for (const amount of amounts) {
+      if (!this.isValidAmount(amount)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  isValidAmount(amount: number): boolean {
+    const bigNumberAmount = new BigNumber(amount);
+    if (!bigNumberAmount.isInteger() || bigNumberAmount.isLessThanOrEqualTo(0)) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Decodes raw tx data into messages, signing info, and fee data
+   * @param txHex - raw base64 tx
+   */
+  getDecodedTxFromRawBase64(txRaw: string): DecodedTxRaw {
+    return decodeTxRaw(fromBase64(txRaw));
+  }
+
+  /**
+   * Returns the array of messages in the body of the decoded transaction
+   * @param decodedTx
+   */
+  private getEncodedMessagesFromDecodedTx(decodedTx: DecodedTxRaw): EncodeObject[] {
+    return decodedTx.body.messages;
+  }
+
+  /**
+   * Arranges a decoded message into our MessageData value interface
+   * @param decodedMessage
+   * @private
+   */
+  private getMessageValueDataFromDecodedMessage(decodedMessage: any): any {
+    return {
+      fromAddress: decodedMessage.fromAddress,
+      toAddress: decodedMessage.toAddress,
+      amount: decodedMessage.amount,
+    };
+  }
+
+  /**
+   * Returns an array of MessageData[] from an encoded messages array
+   * @param encodedMessages
+   * @private
+   */
+  private getMessageDataFromEncodedMessages(encodedMessages: EncodeObject[]): MessageData[] {
+    const registry = new Registry(defaultRegistryTypes);
+    const messageData: MessageData[] = [];
+    for (const message of encodedMessages) {
+      messageData.push({
+        value: this.getMessageValueDataFromDecodedMessage(registry.decode(message)),
+        typeUrl: message.typeUrl,
+      });
+    }
+    return messageData;
+  }
+
+  /**
+   * Pulls the sequence number from a DecodedTxRaw AuthInfo property
+   * @param decodedTx
+   * @private
+   */
+  private getSequenceFromDecodedTx(decodedTx: DecodedTxRaw): number {
+    return Number(decodedTx.authInfo.signerInfos[0].sequence);
+  }
+
+  /**
+   * Pulls the typeUrl from the encoded message of a DecodedTxRaw
+   * @param decodedTx
+   */
+  getTypeUrlFromDecodedTx(decodedTx: DecodedTxRaw): string {
+    const encodedMessage = this.getEncodedMessagesFromDecodedTx(decodedTx)[0];
+    return encodedMessage.typeUrl;
+  }
+
+  getGasBudgetFromDecodedTx(decodedTx: DecodedTxRaw): GasFeeLimitData {
+    return {
+      amount: decodedTx.authInfo.fee?.amount as Coin[],
+      gas: Number(decodedTx.authInfo.fee?.gasLimit),
+    };
+  }
+
+  getExplicitSignerDataFromDecodedTx(decodedTx: DecodedTxRaw): SignerData {
+    const sequence = this.getSequenceFromDecodedTx(decodedTx);
+    return {
```

Comment:
Can you assign ticket numbers for this TODOs?

Reply:
Yes will do shortly